dask/array/core.py

```diff
@@ -1092,9 +1092,21 @@ def keys(*args):
     def __dask_tokenize__(self):
         return self.name
 
-    __dask_optimize__ = globalmethod(
-        optimize, key="array_optimize", falsey=dont_optimize
-    )
+    @staticmethod
+    def __dask_optimize__(dsk, keys, **kwargs):
+        if "array_optimize" in config.config:  # legacy value
+            func = config.get("array_optimize")
```

Comment:
I think we should deprecate the legacy values

Reply:
I agree.  There are currently two PRs dealing with configuration right now.  I'd rather merge both of them and then do the deprecation afterwards if possible.  

Any thoughts on if we should warn or error on the old values?

Reply:
I think we'll want to warn for a release, just to deprecate nicely for anyone using them.

Reply:
OK to hold this off for a follow-up PR?

Reply:
Slight preference to get it in this one just to make sure we deprecate these before the next release. Happy either way though, up to you.

Reply:
OK, I've handled this as requested.  I added it to #6087 though